lib/cli/src/generators/SVELTE/index.ts

```diff
@@ -1,9 +1,47 @@
+import fse from 'fs-extra';
+import { logger } from '@storybook/node-logger';
+
 import { baseGenerator, Generator } from '../baseGenerator';
 
 const generator: Generator = async (packageManager, npmOptions, options) => {
-  baseGenerator(packageManager, npmOptions, options, 'svelte', {
+  await baseGenerator(packageManager, npmOptions, options, 'svelte', {
     extraPackages: ['svelte', 'svelte-loader'],
+    extraAddons: ['@storybook/addon-svelte-csf'],
   });
+
+  let conf = fse.readFileSync('./.storybook/main.js').toString();
+
+  // add *.stories.svelte
+  conf = conf.replace(/js\|jsx/g, 'js|jsx|svelte');
+
+  let requirePreprocessor;
+  let preprocessStatement = 'undefined';
+
+  // svelte.config.js ?
+  if (fse.existsSync('./svelte.config.js')) {
+    logger.info("Configuring preprocessor from 'svelte.config.js'");
+
+    requirePreprocessor = `const preprocess = require("../svelte.config.js").preprocess;`;
+    preprocessStatement = 'preprocess';
+  } else {
+    // svelte-preprocess dependencies ?
+    const packageJson = packageManager.retrievePackageJson();
+    if (packageJson.devDependencies && packageJson.devDependencies['svelte-preprocess']) {
+      logger.info("Configuring preprocessor with 'svelte-preprocess'");
+
+      requirePreprocessor = 'const sveltePreprocess = require("svelte-preprocess");';
+      preprocessStatement = 'sveltePreprocess()';
+    }
+  }
+
+  const svelteOptions = `  "svelteOptions": {\n    preprocess: ${preprocessStatement},\n  },`;
+
+  if (requirePreprocessor) {
+    conf = `${requirePreprocessor}\n\n${conf}`;
+  }
+
+  conf = conf.replace(/\],/, `],\n${svelteOptions}`);
```

Comment:
This is also not very robust. Same as the in the configure method of `lib\cli\src\generators\baseGenerator.ts` you have the capacity to add "extraMain" args to your main. With a small refactor, you won't need to know about `./.storybook/main.js` in this file

Reply:
That's why you're the MVP @tooppaaa ❤️ 

Thanks so much for the thoughtful review!

Reply:
Currently, the main.js is generating from static properties (with JSON.stringify).

However, in my case, I want to inject : 
- An import statement / require
- A Function (which can't be generated by JSON.stringify), so I have to format a string and format it correctly 

```js
const sveltePreprocess = require('svelte-preprocess');

module.exports = {
   ...
   svelteOptions: {
     preprocessor: sveltePreprocess()
  },
  ..
}
```

That's why, for me at least, it wasn't a "small refactor" but I have to rewrite most of the logic inside configure.js 😅 
I didn't found jest test around this function, so this refactoring could be risky

I can work on that, but how do you see the implementation of `configure` for my case ?